infra/main.bicep

```diff
@@ -0,0 +1,751 @@
+
+@description('Is the index prechunked')
+param azureSearchIndexIsPrechunked string = 'false'
+
+@description('Top K results')
+param azureSearchTopK string = '5'
+
+@description('Enable in domain')
+param azureSearchEnableInDomain string = 'false'
+
+@description('Content columns')
+param azureSearchContentColumns string = 'content'
+
+@description('Filename column')
+param azureSearchFilenameColumn string = 'filename'
+
+@description('Title column')
+param azureSearchTitleColumn string = 'title'
+
+@description('Url column')
+param azureSearchUrlColumn string = 'url'
+
+@description('Name of Azure OpenAI Resource')
+param azureOpenAIResourceName string = '${environmentName}-openai-${resourceToken}'
+
+@description('Name of Azure OpenAI Resource SKU')
+param azureOpenAISkuName string = 'S0'
+
+@description('Azure OpenAI Model Deployment Name')
+param azureOpenAIModel string = 'gpt-35-turbo'
+
+@description('Azure OpenAI Model Name')
+param azureOpenAIModelName string = 'gpt-35-turbo'
+
+param azureOpenAIModelVersion string = '0613'
+
+@description('Orchestration strategy: openai_function or langchain str. If you use a old version of turbo (0301), plese select langchain')
+@allowed([
+  'openai_function'
+  'langchain'
+])
+param orchestrationStrategy string = 'langchain'
+
+@description('Azure OpenAI Temperature')
+param azureOpenAITemperature string = '0'
+
+@description('Azure OpenAI Top P')
+param azureOpenAITopP string = '1'
+
+@description('Azure OpenAI Max Tokens')
+param azureOpenAIMaxTokens string = '1000'
+
+@description('Azure OpenAI Stop Sequence')
+param azureOpenAIStopSequence string = '\n'
+
+@description('Azure OpenAI System Message')
+param azureOpenAISystemMessage string = 'You are an AI assistant that helps people find information.'
+
+@description('Azure OpenAI Api Version')
+param azureOpenAIApiVersion string = '2023-07-01-preview'
+
+@description('Whether or not to stream responses from Azure OpenAI')
+param azureOpenAIStream string = 'true'
+
+@description('Azure OpenAI Embedding Model Deployment Name')
+param azureOpenAIEmbeddingModel string = 'text-embedding-ada-002'
+
+@description('Azure OpenAI Embedding Model Name')
+param azureOpenAIEmbeddingModelName string = 'text-embedding-ada-002'
+
+@description('Azure AI Search Resource')
+param azureAISearchName string = '${environmentName}-search-${resourceToken}'
+
+@description('The SKU of the search service you want to create. E.g. free or standard')
+@allowed([
+  'free'
+  'basic'
+  'standard'
+  'standard2'
+  'standard3'
+])
+param azureSearchSku string = 'standard'
+
+@description('Azure AI Search Index')
+param azureSearchIndex string = '${environmentName}-index-${resourceToken}'
+
+@description('Azure AI Search Conversation Log Index')
+param azureSearchConversationLogIndex string = 'conversations'
+
+@description('Name of Storage Account')
+param storageAccountName string = '${environmentName}str${resourceToken}'
```

Comment:
@zedy-wj - Let's do what we do in other azd templates where we don't have issues.  Thanks.  I don't think we use env name in storage account name because that can be very long.